atheris/import_hook.py

```diff
@@ -0,0 +1,140 @@
+        package_name = fullname.split(".")[0]
+        
+        if (not self._target_packages or package_name in self._target_packages) and package_name != "atheris":
+            spec = PathFinder.find_spec(fullname, path, target)
+            
+            if spec is None or spec.loader is None:
+                return None
+            
+            if isinstance(spec.loader, SourceFileLoader):
+                spec.loader = AtherisSourceFileLoader(spec.loader.name, spec.loader.path, self._trace_dataflow)
+            elif isinstance(spec.loader, SourcelessFileLoader):
+                spec.loader = AtherisSourcelessFileLoader(spec.loader.name, spec.loader.path, self._trace_dataflow)
+            else:
+                return None
+            
+            spec.loader_state = None
+            
+            print(f"INFO: Instrumenting {fullname}")
+            
+            return spec
+        
+        else:
+            return None
+    
+    def invalidate_caches(self):
+        return PathFinder.invalidate_caches()
+    
+class AtherisSourceFileLoader(SourceFileLoader):
+    def __init__(self, name, path, trace_dataflow):
+        super().__init__(name, path)
+        self._trace_dataflow = trace_dataflow
+    
+    def get_code(self, fullname):
+        code = super().get_code(fullname)
+        
+        if code is None:
+            return None
+        else:
+            return patch_code(code, self._trace_dataflow)
+    
+class AtherisSourcelessFileLoader(SourcelessFileLoader):
+    def __init__(self, name, path, trace_dataflow):
+        super().__init__(name, path)
+        self._trace_dataflow = trace_dataflow
+    
+    def get_code(self, fullname):
+        code = super().get_code(fullname)
+        
+        if code is None:
+            return None
+        else:
+            return patch_code(code, self._trace_dataflow)
+
+class HookManager:
+    def __init__(self, packages, trace_dataflow):
+        self._target_packages = packages
+        self._trace_dataflow = trace_dataflow
+    
+    def __enter__(self):
+        i = 0
+        while i < len(sys.meta_path):
+            if isinstance(sys.meta_path[i], AtherisMetaPathFinder):
+                return self
+            i += 1
+        
+        i = 0
+        while i < len(sys.meta_path) and sys.meta_path[i] in [BuiltinImporter, FrozenImporter]:
+            i += 1
+        
+        sys.meta_path.insert(i, AtherisMetaPathFinder(self._target_packages, self._trace_dataflow))
+        
+        return self
+        
+    def __exit__(self, *args):
+        i = 0
+        while i < len(sys.meta_path):
+            if isinstance(sys.meta_path[i], AtherisMetaPathFinder):
+                sys.meta_path.pop(i)
+            else:
+                i += 1
+
+def instrument(*modules, trace_dataflow=True):
+    """
+    This function temporarily installs an import hook which instruments
+    all imported modules.
+    The arguments to this function are names of modules or packages.
```

Comment:
Doesn't actually say what those module names do.

I'd actually recommend taking two arguments: `include` and `exclude`. Specifying both wouldn't make sense, unless we end up supporting globs or some sort of hierarchical specification someday. (e.g. include all of `foo.bar` except `foo.bar.baz`.